### ISSUE TITLE
Is it safe to compare against Double.Nan?

### ISSUE THREAD

**Issue by isaacabraham** [Zaid-Ajaj/Feliz#461]
The code for `prop.onChange` is [here](https://github.com/Zaid-Ajaj/Feliz/blob/6350ef036d7f49c9f70b4fb56e93156d5569ada8/Feliz/Properties.fs#L1074). I don't know how this gets rendered in the JS world but in the .NET world IIRC you shouldn't compare against the `Nan` value but instead use the `IsNan` method.

For example (and I tested this in Fable too), the following behaviour can be observed:

```fsharp
Browser.Dom.console.log $"Nan EQ Check: {Double.NaN = Double.NaN}" // false
Browser.Dom.console.log $"Nan NEQ Check: {Double.NaN <> Double.NaN}" // true
Browser.Dom.console.log $"Nan IsNan Check: {Double.IsNaN Double.NaN}" // true
```

In other words, I think that that NaN check in the line linked above will never return false, and should either be removed or replaced with `IsNan`. If so, let me know and I'll raise a PR.

**Comment by tnishimura** [comment]
Indeed, according to IEEE 754, "any_double != NaN" should eval to true. [Useful table here](https://en.wikipedia.org/wiki/NaN#Comparison_with_NaN).

**Comment by Zaid-Ajaj** [comment]
You are right @isaacabraham, using `IsNaN` is the correct way for checking. Resolved as of v1.60
